src/python/dxpy/templating/templates/nextflow/src/nextflow.sh

```diff
@@ -69,29 +69,161 @@ on_exit() {
     set -xe
   fi
 
+  # backup cache
+  if [[ $preserve_cache == true ]]; then
+    echo "=== Execution complete — caching current session to $DX_CACHEDIR/$NXF_UUID"
+
+    # wrap cache folder and upload cache.tar
+    if [[ -n "$(ls -A .nextflow)" ]]; then
+      tar -cf cache.tar .nextflow
+
+      CACHE_ID=$(dx upload "cache.tar" --path "$DX_CACHEDIR/$NXF_UUID/cache.tar" --no-progress --brief --wait -p -r) &&
+        echo "Upload cache of current session as file: $CACHE_ID" &&
+        rm -f cache.tar ||
+        echo "Failed to upload cache of current session $NXF_UUID"
+    else
+      echo "No cache is generated from this execution. Skip uploading cache."
+    fi
 
-  # remove .nextflow from the current folder /home/dnanexus/output_files
+  # preserve_cache is false
+  # clean up files of this session
+  else
+    echo "=== Execution complete — cache and working files will not be resumable"
+  fi
+
+  # remove .nextflow from the current folder /home/dnanexus/nextflow_playground
   rm -rf .nextflow
+  rm nxf_runtime.config
 
   # try uploading the log file if it is not empty
   if [[ -s $LOG_NAME ]]; then
-    mkdir ../nextflow_log
-    mv $LOG_NAME ../nextflow_log/$LOG_NAME || true
+    mkdir -p /home/dnanexus/out/nextflow_log
+    mv "$LOG_NAME" "/home/dnanexus/out/nextflow_log/$LOG_NAME" || true
   else
     echo "No nextflow log file available."
   fi
-  
+
   # upload the log file and published files if any
-  cd ..
-  if [[ -d ./nextflow_log || -n "$(ls -A ./output_files)" ]]; then
-    dx-upload-all-outputs --parallel || true
-  else
-    echo "No log file or output files has been generated."
-  fi
+  mkdir -p /home/dnanexus/out/published_files
+  find . -type f -newermt "$BEGIN_TIME" -exec mv {} /home/dnanexus/out/published_files/ \;
+
+  dx-upload-all-outputs --parallel --wait-on-close || echo "No log file or published files has been generated."
   # done
   exit $ret
 }
 
+get_resume_session_id() {
+  if [[ $resume == 'true' || $resume == 'last' ]]; then
+    # find the latest job run by applet with the same name
+    echo "Will try to find the session ID of the latest session run by $EXECUTABLE_NAME."
+    PREV_JOB_DESC=$(dx api system findExecutions \
+      '{"state":["done","failed"],
+    "created": {"after": 1.5552e10},
```

Comment:
Shouldn't we provide now - 6 months value here?

Reply:
See my comment above, thanks!

Reply:
Fixed